processFlows.dscp

```diff
@@ -43,3 +50,12 @@ pub fn issue_cert |input: InitiatedCert| => |output: IssuedCert| where {
   output.hydrogen_owner != sender,
   output.energy_owner == sender,
 }
+
+pub fn revoke_cert |input: IssuedCert| => |output: RevokedCert| where {
+  output == input,
+  input.hydrogen_owner == output.hydrogen_owner,
+  input.energy_owner == output.energy_owner,
+  output.energy_owner != sender,
+  output.hydrogen_owner != sender,
+  output.regulator == sender,
```

Comment:
this isn't quite enough. We need to know who the regulator for this certficate is prior to it being issued. You'll need to update the `initiate_cert` and `issue_cert` flows to include a `regulator` and assert in each of the other flows that they are persisted. This will also lead to `ts` changes to include the regulator persona

Reply:
Thanks for a flag up, have had this suspicion. This has been updated to include regulator along with some basic checks. When you get a moment please confirm.